internal/textutil/diff.go

```diff
@@ -0,0 +1,74 @@
+package textutil
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Diff returns a formatted diff of the two texts,
+// showing the entire text and the minimum line-level
+// additions and removals to turn text1 into text2.
+// (That is, lines only in text1 appear with a leading -,
+// and lines only in text2 appear with a leading +.)
+func Diff(text1, text2 string) string {
+	if text1 != "" && !strings.HasSuffix(text1, "\n") {
+		text1 += "(missing final newline)"
+	}
+	lines1 := strings.Split(text1, "\n")
+	lines1 = lines1[:len(lines1)-1] // remove empty string after final line
+	if text2 != "" && !strings.HasSuffix(text2, "\n") {
+		text2 += "(missing final newline)"
+	}
+	lines2 := strings.Split(text2, "\n")
+	lines2 = lines2[:len(lines2)-1] // remove empty string after final line
+
+	// Naive dynamic programming algorithm for edit distance.
+	// https://en.wikipedia.org/wiki/Wagner–Fischer_algorithm
+	// dist[i][j] = edit distance between lines1[:len(lines1)-i] and lines2[:len(lines2)-j]
+	// (The reversed indices make following the minimum cost path
+	// visit lines in the same order as in the text.)
+	dist := make([][]int, len(lines1)+1)
```

Comment:
FYI, there are ways to implement this in non-quadratic memory. Shouldn't matter if the inputs will always be relatively small, though.

Reply:
This is literally a copy-paste of Russ' implementation. I don't think, per his comment, he was looking to win any awards with his initial implementation.

Reply:
Gotcha :) this was just a drive-by comment, in case you had written this yourself. Hence my previous comment about whether this package was a copy-paste.